src/main/java/seedu/address/model/person/RecurringSchedule.java

```diff
@@ -37,35 +37,80 @@ public class RecurringSchedule extends Schedule {
      */
     public RecurringSchedule(String schedule) {
         super(validateThenExtractStartTime(schedule), extractEndTime(schedule));
+        requireNonNull(schedule);
+        assert isValidRecurringSchedule(schedule) : "Schedule should be valid by this point";
         this.day = extractDay(schedule);
     }
+    /**
+     * Validates the schedule string and extracts the start time if valid.
+     *
+     * @param schedule The schedule string to validate and extract from.
+     * @return The extracted start time.
+     * @throws NullPointerException if schedule is null.
+     * @throws IllegalArgumentException if schedule format is invalid.
+     */
     private static String validateThenExtractStartTime(String schedule) {
         requireNonNull(schedule);
         checkArgument(isValidRecurringSchedule(schedule), MESSAGE_CONSTRAINTS);
         return extractStartTime(schedule);
     }
 
+    /**
+     * Extracts the day of week from a valid schedule string.
+     *
+     * @param schedule A valid schedule string.
+     * @return The extracted day of week.
+     */
     private static DayOfWeek extractDay(String schedule) {
-        return DayOfWeekUtils.fromString(schedule.split(" ")[0]);
+        String[] parts = schedule.split(" ");
+        assert parts.length >= 1 : "Schedule string should have at least day part";
+        return DayOfWeekUtils.fromString(parts[0]);
     }
 
+    /**
+     * Extracts the start time from a valid schedule string.
+     *
+     * @param schedule A valid schedule string.
+     * @return The extracted start time.
+     */
     private static String extractStartTime(String schedule) {
-        return schedule.split(" ")[1];
+        String[] parts = schedule.split(" ");
+        assert parts.length >= 2 : "Schedule string should have at least day and start time parts";
+        return parts[1];
     }
 
+    /**
+     * Extracts the end time from a valid schedule string.
+     *
+     * @param schedule A valid schedule string.
+     * @return The extracted end time.
+     */
     private static String extractEndTime(String schedule) {
-        return schedule.split(" ")[2];
+        requireNonNull(schedule);
+        String[] parts = schedule.split(" ");
```

Comment:
```suggestion
        String[] parts = schedule.split(" ");
```
Similar to above change.